src/main/java/core/basesyntax/strategy/impl/PurchaseOperationImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperationImpl implements OperationHandler {
+    @Override
+    public void applyOperation(FruitTransaction transaction) {
+        if (transaction.getQuantity() <= MIN_QUANTITY) {
+            throw new RuntimeException("Purchase must be 1 or more items. But was: "
+                    + transaction.getQuantity());
+        }
+        if ((getCurrentBalance(transaction) - transaction.getQuantity()) < 0) {
```

Comment:
The method getCurrentBalance() is called but not defined in this class. Ensure that you have defined this method or that it is correctly imported from another class.